packages/x-date-pickers/src/AdapterMomentHijri/AdapterMomentHijri.ts

```diff
@@ -213,27 +213,6 @@ export class AdapterMomentHijri extends AdapterMoment implements MuiPickersAdapt
     return value.clone().iDate(date);
   };
 
-  public getWeekArray = (value: Moment) => {
-    const start = value.clone().startOf('iMonth').startOf('week');
-
-    const end = value.clone().endOf('iMonth').endOf('week');
-
-    let count = 0;
-    let current = start;
-    const nestedWeeks: Moment[][] = [];
-
-    while (current.isBefore(end)) {
-      const weekNumber = Math.floor(count / 7);
-      nestedWeeks[weekNumber] = nestedWeeks[weekNumber] || [];
-      nestedWeeks[weekNumber].push(current);
-
-      current = current.clone().add(1, 'day');
-      count += 1;
-    }
-
-    return nestedWeeks;
```

Comment:
The only reason why we can't remove `getWeekArray` from the adapter and move it to a utility method is the locale sync made by all adapters except `date-fns` which already passes the locale in `startOfWeek`.

The abstraction might be slightly better if we remove this method and instead expose `setLocaleToValue`.

The less complex abstraction we have in the adapter, the better imho.